hibernate-core/src/main/java/org/hibernate/cache/internal/CacheKeyImplementation.java

```diff
@@ -55,6 +55,7 @@
 
 	private int calculateHashCode(Type type, SessionFactoryImplementor factory) {
 		int result = type.getHashCode( id, factory );
+		result = 31 * result + entityOrRoleName.hashCode();
```

Comment:
Hmm, I thought that entity name is already handled in `type.getHashCode`. What is the difference between `Type` and entityOrRoleName?

Reply:
No, the entity name is not handled by ```type.getHashCode```.

The ```type``` argument is the ID type. So, if the ID is an ```int```, then its ```Type`` is ```org.hibernate.type.IntegerType```. The hashCode is ultimately the hashCode of the ID, computed [here](https://github.com/hibernate/hibernate-orm/blob/master/hibernate-core/src/main/java/org/hibernate/type/descriptor/java/AbstractTypeDescriptor.java#L76-L79).